### PR TITLE
fix: only call `/api/me/votes` when being signed in

### LINKED ISSUE #522: todo(perf): avoid calling `GET /api/me/votes` if user is not signed in
This call is unnecessary in this case.

### PR DESCRIPTION
**Pull request by mbhrznr**
closes #522.

### CODE REVIEW

File: islands/ItemsList.tsx [hunk 1]
```diff
@@ -61,12 +63,17 @@ export default function ItemsList(props: { endpoint: string }) {
...
+      return;
+    }
+
     fetchVotedItems()
       .then((votedItems) =>
         votedItemsIdsSig.value = votedItems.map(({ id }) => id)
       )
       .finally(() => loadMoreItems());
-  }, []);
+  }, [props.isSignedIn]);
```
Comment (iuioiua) [inline islands/ItemsList.tsx:76]: Is this needed? I'm not sure it is.

Comment (iuioiua) [reply islands/ItemsList.tsx]: I removed this as I think it's unnecessary. If you think it's necessary, please let me know.

**Review by iuioiua — APPROVED** [review]
I've made some edits and now, LGTM! Thank you, Michael.

File: islands/ItemsList.tsx (hunk 1 above)
Comment (mbhrznr) [reply islands/ItemsList.tsx]: shouldn't be needed, thanks for removing!

**Pull request merged**